src/event.js

```diff
@@ -91,6 +91,36 @@ function on( elem, types, selector, data, fn, one ) {
 	} );
 }
 
+function addEventGetter( unused, name ) {
+	Object.defineProperty( jQuery.Event.prototype, name, {
+		enumerable: true,
+		configurable: true,
+
+		get: function() {
+			var value, hook;
+
+			if ( this.originalEvent ) {
+				if ( hook = jQuery.event.propHooks[ name ] ) {
+					value = hook( this.originalEvent );
+				} else {
+					value = this.originalEvent[ name ];
+				}
+			}
+
+			return this[ name ] = value;
+		},
+
+		set: function( value ) {
+			Object.defineProperty( this, name, {
```

Comment:
I've removed the `this[ name ] = value` in the getter, but kept the setter to allow things like unit tests to override the getter.  This should probably still be benchmarked.  I don't have any benchmarks setup right now. If anyone else wants to give it a shot then feel free! Otherwise I'll try to find some time at some point...